src/components/HomePage/SearchArea/SearchArea.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { connect } from 'react-redux';
+import { addSnackbar } from '../../../actions/notificationActions';
+
+import JOB_TYPES from './jobTypes';
+
+import FormControl from '@material-ui/core/FormControl';
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import Slider from '@material-ui/core/Slider';
+import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import Collapse from '@material-ui/core/Collapse';
+
+import { makeStyles } from '@material-ui/styles';
+import searchAreaStyle from "./SearchArea.module.css";
+
+import SearchBar from "./SearchBar";
+import ShowAdvancedOptionsButton from './ShowAdvancedOptionsButton';
+import SliderValueTooltip from './SliderValueTooltip';
+
+const INITIAL_JOB_TYPE = "";
+const INITIAL_JOB_DURATION = 1;
+
+const useStyles = makeStyles({
+    wrapperInner: {
+        display: "grid",
+        "grid-template-columns": "1fr 1fr",
+        "grid-template-rows": "1fr",
+        "align-items": "center",
+        "grid-gap": "1em",
+    }
+});
+
+function SearchArea(props) {
+
+    const {addSnackbar, onSubmit} = props;
+    const classes = useStyles();
+    
+    // Set initial form values
+    const [searchValue, setSearchValue] = React.useState("");
+    const [advancedSearch, setAdvancedSearch] = React.useState(false);
+    const [jobType, setJobType] = React.useState(INITIAL_JOB_TYPE);
+    const [jobDuration, setJobDuration] = React.useState(INITIAL_JOB_DURATION);
+    
+    const toggleAdvancedOptions = () => {
```

Comment:
This seems extractable to a `useToggledState` or similarly-named hook, exporting something like `[state, toggleState]`

Reply:
This PR is undergoing a transition into the new Hooks API + Functional Components. I believe all the components logic could be improved, so do not let me forget that suggestion

Reply:
The transition can be a bit more gradual and worked on later, if need be - I'm fine with that, the React docs themselves say that that's ok.
As for a reminder, you can just leave this conversation open, but I'd say something in the final code review too, probably 😜

Reply:
Created a custom hook `useToggle` that returns `[state, toggleState]` and hides the `state = !state` logic, so the developer only needs to call `toggleState()` to update the value

Reply:
Sounds perfect! Looking forward to reviewing everything again after a while! 🙃 😄